doc/source/reader_table.py

```diff
@@ -20,6 +20,11 @@
 
 from satpy.readers import available_readers
 
+try:
+    from yaml import BaseLoader
+except ImportError:
+    from yaml import BaseLoader  # type: ignore
```

Comment:
I'm not sure how this can work? this is importing twice the same thing, right?

Reply:
Indeed. I removed to much from the second import I guess. Apart from that this is something I forgot to add to the PR. I think this try/except was introduced in Satpy during the transition phase of pyyaml. The first import works since pyyaml version 5.1 which is already 2 years old. My suggestion would be to update the Satpy requirements to need pyyaml>= 5.1 then we could do away with the try/except.

Reply:
I can add a test for this if it is decided that this indeed should be the solution (@djhoese mentioned something about running the table generating function inside the mocking but that seemed a little more complicated to me and I don't know if that is even possible).

Reply:
yes, I'm good with setting a limit on pyyaml version

Reply:
I actually have a special loader in this test code that ignores all tags so that imports can be skipped over:

https://github.com/pytroll/satpy/blob/267e779cea71cb5286d5edc1b4da7478eb62fc62/satpy/tests/test_readers.py#L632-L635

If that isn't needed and the BaseLoader can be used that'd be pretty nice.

Reply:
It seems the BaseLoader works. I will add a test tomorrow.

Reply:
Yes, the base loader ignore anything python-specific iirc